csharp/ql/lib/semmle/code/csharp/dataflow/internal/FlowSummaryImplSpecific.qll

```diff
@@ -223,3 +221,19 @@ predicate interpretInputSpecific(string c, InterpretNode mid, InterpretNode n) {
     a.getUnboundDeclaration() = mid.asElement()
   )
 }
+
+/** Gets the argument position obtained by parsing `X` in `Parameter[X]`. */
+bindingset[s]
+ArgumentPosition parseParamBody(string s) {
+  result.getPosition() = s.regexpCapture("([-0-9]+)", 1).toInt()
```

Comment:
Seems cleaner to have a separate private predicate that does the regex parsing from string to int, and then have `parseParamBody` and `parseArgBody` both use that.